src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionScopeInternal.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class ClientConnectionScopeProperties
+    {
+        public IServiceConnection ServiceConnection { get; set; }
+
+        // todo: add additional properties to flow with the scope here
+    }
+
+    internal class ServiceConnectionScopeHolder
+    {
+        private static AsyncLocal<ServiceConnectionScopeHolder> s_AsyncLocalSCC = new AsyncLocal<ServiceConnectionScopeHolder>();
+        internal static ServiceConnectionScopeHolder Current
+        {
+            get => s_AsyncLocalSCC.Value;
+            set => s_AsyncLocalSCC.Value = value;
+        }
+
+        public ClientConnectionScopeProperties Properties { get; set; }
+    }
+
+    internal class ServiceConnectionScopeInternal : IDisposable
```

Comment:
naming: ServiceConnectionScope~~Internal~~?

Reply:
The name is misleading since it's now designed to contain more properties than service connection.

Reply:
Agree - the scope is not even service bound.

Reply:
This is now `ClientConnectionScopeInternal`.
It carries the `ClientConnectionScopeProperties` with `IServiceConnection OutboundServiceConnection` and more properties in the future.